src/source/Signaling/Signaling.c

```diff
@@ -328,6 +342,9 @@ STATUS signalingSendMessageSync(PSignalingClient pSignalingClient, PSignalingMes
                               pSignalingMessage->payload, pSignalingMessage->payloadLen,
                               pSignalingMessage->correlationId, 0));
 
+    // Update the internal diagnostics only after successfully sending
+    ATOMIC_INCREMENT(&pSignalingClient->diagnostics.numberOfMessagesSent);
+
 CleanUp:
 
     CHK_LOG_ERR(retStatus);
```

Comment:
I think we should check the error here, and increment `numberOfRuntimeErrors`

Reply:
Not necessary for now as the sending of the messages is a synchronous operation and the error will be returned in the call itself making it not a runtime error but rather an operation failure. I will be thinking a little more about the types of errors to add - something like general errors which will account for any operation failures to increment the counter. Not sure how useful that will be though. Perhaps it makes sense, let me do that